src/dialogs/more-info/controls/more-info-camera.ts

```diff
@@ -34,14 +36,51 @@ class MoreInfoCamera extends LitElement {
         allow-exoplayer
         controls
       ></ha-camera-stream>
+
+      <div class="actions">
+        <ha-progress-button
+          @click=${this._takeSnapshot}
+          .disabled=${this.stateObj.state === UNAVAILABLE}
+        >
+          ${this.hass.localize(
+            "ui.dialogs.more_info_control.camera.take_snapshot"
+          )}
+        </ha-progress-button>
+      </div>
     `;
   }
 
+  private async _takeSnapshot() {
+    const progressElement =
+      this.shadowRoot!.querySelector("ha-progress-button")!;
+
+    try {
+      await this.hass!.callService("camera", "snapshot", {
+        entity_id: this.stateObj!.entity_id,
+      });
```

Comment:
I wonder if this is the expected thing to happen...

I maybe would think I could download a snapshot with that button, while is actually stores one in the media folder.

Reply:
Hmm yes, good point. The more I think about it, the more I think it would be better to download the snapshot instead of saving it. Or would it make sense to have both options?

Reply:
Honestly, not sure. I think this would be more of a UX/product question. Let me drop this with the P&D team.